rudder-lang/libs/oslib.rl

```diff
@@ -88,13 +111,275 @@ items in solaris {
   *
 }
 
+# == suse
+# TODO should we use suse or sles ?
+items in sles {
+  sles_10,
+  sles_11,
+  sles_12,
+  sles_15,
+  *
+}
+
+items in sles_10 {
+  sles_10_0,
+  sles_10_1,
+  sles_10_2,
+  sles_10_3,
+  sles_10_4,
+}
+
+items in sles_11 {
+  sles_11_0,
+  sles_11_1,
+  sles_11_2,
+  sles_11_3,
+}
+
+items in sles_12 {
+  sles_12_0,
+  sles_12_1,
+  sles_12_2,
+  sles_12_3,
+  sles_12_4,
+  *
+}
+
+items in sles_15 {
+  sles_15_0,
+  sles_15_1,
+  *
+}
+
+# == sles: version checked (cfengine names not checked)
+
+# == centos
+items in centos {
+  centos_3,
+  centos_4,
+  centos_5,
+  centos_6,
+  centos_7,
+  centos_8,
+  *
+}
+
+# == centos: cfengine checked, no version check
+
+# == redhat
+# TODO should we use redhat or rhel ?
+items in redhat {
+  redhat_3,
+  redhat_4,
+  redhat_5,
+  redhat_6,
+  redhat_7,
+  redhat_8,
+  *
+}
+
+items in redhat_3 {
+  redhat_3_0,
+  redhat_3_1,
+  redhat_3_2,
+  redhat_3_3,
+  redhat_3_4,
+  redhat_3_5,
+  redhat_3_6,
+  redhat_3_7,
+  redhat_3_8,
+  redhat_3_9,
+}
+
+items in redhat_4 {
+  redhat_4_0,
+  redhat_4_1,
+  redhat_4_2,
+  redhat_4_3,
+  redhat_4_4,
+  redhat_4_5,
+  redhat_4_6,
+  redhat_4_7,
+  redhat_4_8,
+  redhat_4_9,
+}
+
+items in redhat_5 {
+  redhat_5_0,
+  redhat_5_1,
+  redhat_5_2,
+  redhat_5_3,
+  redhat_5_4,
+  redhat_5_5,
+  redhat_5_6,
+  redhat_5_7,
+  redhat_5_8,
+  redhat_5_9,
+  redhat_5_10,
+  redhat_5_11,
+}
+
+items in redhat_6 {
+  redhat_6_0,
+  redhat_6_1,
+  redhat_6_2,
+  redhat_6_3,
+  redhat_6_4,
+  redhat_6_5,
+  redhat_6_6,
+  redhat_6_7,
+  redhat_6_8,
+  redhat_6_9,
+  redhat_6_10,
+  *
+}
+
+items in redhat_7 {
+  redhat_7_0,
+  redhat_7_1,
+  redhat_7_2,
+  redhat_7_3,
+  redhat_7_4,
+  redhat_7_5,
+  redhat_7_6,
+  redhat_7_7,
+  *
+}
+
+items in redhat_8 {
+  redhat_8_0,
+  redhat_8_1,
+  *
+}
+
+# == redhat: cfengine + wikipedia checked
+
+# == ubuntu
+# TODO Should we have major only for ubuntu ?
```

Comment:
18_04 is not major+minor but a major version by itself. There is no particular link between 18_04 and 18_10, so I don't think so.

Reply:
I agree